one_fm/purchase/doctype/quotation_comparison_sheet/quotation_comparison_sheet.js

```diff
@@ -139,3 +213,353 @@ var set_quotation_item_details = function(frm, item, quotation) {
+		item => item.quotation === ordered_quotations.quotation
+	);
+	// append to selected filtered table
+	complete_filters_table(frm, best_quotation_items, 'Best Rate from One Supplier');
+}
+
+
+//  filter for best price by many supplier
+let best_price_many_supplier = (frm)=>{
+	get_quotation_items(frm).forEach((item, i) => {
+		make_rfq_dataset_itemsfilter(frm, item, 'Best Rate');
+	});
+	// select best items price
+	let items = get_quotation_items(frm);
+
+	let best_quotation_items = []
+	items.forEach((item, i) => {
+		best_quotation_items.push(
+			frm.doc.quotation_items.filter((a, b) => {
+			    return a.item_name===item
+			}).sort((x, y)=> {
+			    return x.rate - y.rate
+			})[0]
+		)
+	});
+
+	// // append to selected filtered table
+	complete_filters_table(frm, best_quotation_items, 'Best Rate from Many Supplier');
+}
+
+
+//  filter based on earliest delivery
+let earliest_delivery = (frm)=>{
+	get_quotation_items(frm).forEach((item, i) => {
+		make_rfq_dataset_itemsfilter(frm, item, 'Earliest Delivery');
+	});
+	// select earliest delivery
+	let ordered_quotations = frm.doc.quotations.sort((a, b) => {
+	    return new Date(a.estimated_delivery_date) - new Date(b.estimated_delivery_date);
+	})[0];
+	// filter all items for selected quotation
+	let best_quotation_items = frm.doc.quotation_items.filter(
+		item => item.quotation === ordered_quotations.quotation
+	);
+	// append to selected filtered table
+	complete_filters_table(frm, best_quotation_items, 'Earliest Delivery');
+}
+
+
+// custom filter
+let customer_filter = (frm)=>{
+	let items = get_quotation_items(frm);
+	const table_fields = [
+			{
+				fieldname: "item_name", fieldtype: "Select",
+				in_list_view: 1, label: "Quotation Item",
+				options: items, reqd: 1,
+				change: function (x) {
+					dialog.fields_dict.items_detail.df.data.some(d => {
+						if (d.item_name==this.doc.item_name && d.idx != this.doc.idx) {
+
+							this.doc.item_name = null;
+							dialog.fields_dict.items_detail.grid.refresh();
+							return frappe.utils.play_sound("error");
+							frappe.throw('You cannot repeat same item')
+							// d.opening_amount = this.value;
+							return true;
+						}
+					});
+				}
+			},
+			{
+				fieldname: "select_by", fieldtype: "Select",
+				in_list_view: 1, label: "Select by", reqd:1,
+				options: ['Best Rate', 'Earliest Delivery Date'],
+				default: null
+			}
+		];
+
+	const dialog = new frappe.ui.Dialog({
+			title: __('Custom Quotation Selection'),
+			static: false,
+			fields: [
+				// {
+				// 	fieldtype: '', label: __('Company'), default: frappe.defaults.get_default('company'),
+				// 	options: 'Company', fieldname: 'company', reqd: 1
+				// },
+				// {
+				// 	fieldtype: 'Link', label: __('POS Profile'),
+				// 	options: 'POS Profile', fieldname: 'pos_profile', reqd: 1,
+				// 	get_query: () => pos_profile_query,
+				// 	onchange: () => fetch_pos_payment_methods()
+				// },
```

Comment:
Remove unused codes

Reply:
Unused code removed and customer_filter renamed to custom filter but please close the PR, the PR was meant for staging not master. I will open new PR